src/models/wallet.js

```diff
@@ -0,0 +1,55 @@
+var StellarBase = require('stellar-base');
+var Ed25519 = require('ed25519');
+var Sjcl = require('sjcl');
+
+var DbUtil = require('../utils/db');
+var CryptoUtil = require('../utils/crypto');
+
+var wallet = {
+    
+    create: function(msisdn, pin) {
+        var salt = Sjcl.codec.base64.fromBits(Sjcl.random.randomWords(64/4));
+        var key = this._generateKeyPair();
+        var privateKey = key.privateKey;
+        var publicKey = key.publicKey;
+        var encryptionKey = pin;
+        var pinHash = CryptoUtil.hash(pin, salt);
+        var privateKeyEncrypted = CryptoUtil.encryptData(privateKey, encryptionKey);
+        var address = this.addressFromPublicKey(publicKey);
+
+        return DbUtil.promiseConnection()
+            .then(DbUtil.chainedQuery(
+                "INSERT INTO wallet ( msisdn, address, publickey, privatekey, pinhash, salt) VALUES ( $1, $2, $3, $4, $5, $6)",
+                [msisdn, pinHash, address, publicKey, privateKeyEncrypted, salt]))
+            .then(function(successResult) {
+                return {
+                    success: true,
+                    publicKey: publicKey,
+                    privateKey: privateKey,
+                    address: address
+                };
+            }, function(failureResult) {
+                return {
+                    success: false
```

Comment:
We probably want to add something here that gives a hint as to why it failed, I suspect the `failureResult` will have some kind of reason buried in it somewhere?